Wikipedia/Code/ExploreViewController.swift

```diff
@@ -0,0 +1,277 @@
+    }
+    
+    public func didTapSettingsButton(_ sender: UIBarButtonItem) {
+        showSettings()
+    }
+    
+    private func updateNavigationBar() {
+        updateNavigationBar(newOffset: abs(extendNavBarViewTopSpaceConstraint.constant))
+    }
+    
+    private func updateNavigationBar(newOffset extNavBarOffset: CGFloat) {
+        let extNavBarHeight = extendedNavBarView.frame.size.height
+        let percentHidden: CGFloat = extNavBarOffset / extNavBarHeight
+        self.searchBarButtonItem?.alpha = percentHidden
+        self.shortTitleButton?.alpha = percentHidden
+        self.longTitleButton?.alpha = 1 - percentHidden
+    }
+    
+    // MARK: - Actions
+    
+    public func showSettings() {
+        let settingsContainer = UINavigationController(rootViewController: WMFSettingsViewController.init(dataStore: self.userStore))
+        present(settingsContainer, animated: true, completion: nil)
+    }
+    
+    public func titleBarButtonPressed() {
+        self.collectionViewController.collectionView?.setContentOffset(CGPoint.zero, animated: true)
+        self.showSearchBar(animated: true)
+    }
+    
+    // MARK: - WMFExploreCollectionViewControllerDelegate
+    
+    func updateSearchBar(offset: CGFloat, animated: Bool)
+    {
+        if (animated) {
+            self.view.layoutIfNeeded() // Apple recommends you call layoutIfNeeded before the animation block ensure all pending layout changes are applied
+            UIView.animate(withDuration: 0.3, delay: 0, usingSpringWithDamping: 0.8, initialSpringVelocity: 0, options: [.allowUserInteraction, .beginFromCurrentState], animations: {
+                self.extendNavBarViewTopSpaceConstraint.constant = -offset
+                // self.searchBarButtonItem?.alpha = newAlpha
+                self.updateNavigationBar(newOffset: offset)
+                self.view.layoutIfNeeded() // layoutIfNeeded must be called from within the animation block
+            });
+        } else {
+            self.extendNavBarViewTopSpaceConstraint.constant = -offset
+            self.updateNavigationBar(newOffset: offset)
+        }
+    }
+    
+    public func showSearchBar(animated: Bool) {
+        updateSearchBar(offset: 0, animated: animated)
+    }
+    
+    public func hideSearchBar(animated: Bool) {
+        let extNavBarHeight = extendedNavBarView.frame.size.height
+        updateSearchBar(offset: extNavBarHeight, animated: animated)
+    }
+    
+    func exploreCollectionViewController(_ collectionVC: WMFExploreCollectionViewController, willBeginScrolling scrollView: UIScrollView) {
+        isUserScrolling = true
+    }
+    
+    func exploreCollectionViewController(_ collectionVC: WMFExploreCollectionViewController, didScroll scrollView: UIScrollView) {
+        //DDLogDebug("scrolled! \(scrollView.contentOffset)")
+        
+        guard self.view != nil else {
```

Comment:
I think that was some weirdness with the original segue embed technique I was doing. I think it can be removed